src/spectroscopy_bluesky/i20/ion_autofill.py

```diff
@@ -0,0 +1,139 @@
+import enum
+
+import bluesk.plan_stubs as bps
+from dodal.common.types import MsgGenerator
+
+from spectroscopy_bluesky.i20.devices.gas_rig import GasInjectionController
+
+# todo are those parameters? those should be generic
+ionchamber2fill = "I0"  # I0, I1, iT, iRef
+targetPressureAr = 35  # mbar
+targetPressureHe = 1800  # mbar
+
+
+class GasToInject(enum.Enum):
+    ARGON = "argon"
+    HELIUM = "helium"
+    KRYPTON = "krypton"
+    NITROGEN = "nitrogen"
+
+
+class IonChamberToFill(enum.Enum):
+    I0 = "I0"
+    I1 = "I1"
+    iT = "iT"
+    iRef = "iRef"
+
+
+def get_gas_valve(gir, gas_to_inject: GasToInject):
+    gas_map = {
+        GasToInject.ARGON: gir.ar_valve,
+        GasToInject.HELIUM: gir.he_valve,
+        GasToInject.KRYPTON: gir.kr_valve,
+        GasToInject.NITROGEN: gir.n2_valve,
+    }
+    return gas_map[gas_to_inject]
```

Comment:
Could: I would probably put this mapping (and the enum) in the device so that you can do something like `abs_set(gir.valve[ARGON], ...)`